helm.tf

```diff
@@ -0,0 +1,35 @@
+resource "helm_release" "ingress" {
+  name       = "ingress-nginx"
+  repository = "https://kubernetes.github.io/ingress-nginx"
+  chart      = "ingress-nginx"
+
+  depends_on = [
+    kubernetes_service.test

```

Comment:
I don't think you technically need this.  The `Ingress` resource can live in the Kubernetes cluster even if the target service doesn't exist.  The Kubernetes API Server (and NGINX Ingress Controller) will eventually get everything up-to-sync if they deploy out-of-order.